### PR TITLE
delete SQL attachments with forms

### PR DESCRIPTION
**Pull request by snopoke**
@calellowitz 

cc @dimagi/scale-team @millerdev

### CODE REVIEW

File: corehq/form_processor/backends/sql/dbaccessors.py [hunk 1]
```diff
@@ -224,13 +225,49 @@ def form_exists(form_id, domain=None):
             return result.form_exists
 
     @staticmethod
-    @transaction.atomic
```
Comment (emord) [inline corehq/form_processor/backends/sql/dbaccessors.py]: Why did you remove the atomic decorator? Would it make sense to move this to the block where forms are deleted?

Comment (snopoke) [inline corehq/form_processor/backends/sql/dbaccessors.py]: Removed it from here since there is only one SQL call so atomicity is implicit. 

We could add it to the `hard_delete_case_and_forms` function which is where this get's called from. Atomicity isn't respected in a sharded environment though since transactions only apply to the proxy DB.

**Comment by millerdev** [comment]
👍 on ✅

**Comment by millerdev** [comment]
can't merge

**Comment by snopoke** [comment]
:white_check_mark:
